src/Export/VTK.jl

```diff
@@ -111,46 +171,167 @@ function component_names(::Type{S}) where S
+
+    for name in fieldnames
+        data = _evaluate_at_grid_nodes(dh, u, name, #=vtk=# Val(true))
+        _vtk_write_nodedata(vtkfile, data, string(name, suffix))
+    end
+    return nothing
+end
+
+"""
+    write_solution(vtks::VTKStream, u::Vector, suffix="")
+
+Save the values at the nodes in the degree of freedom vector `u` to a 
+the a vtk file for each field in `DofHandler` in `vtks`. 
+`suffix` can be used to append to the fieldname.
+`vtks` is typically created by the [`open_vtk`](@ref) function. 
+
+`u` can also contain tensorial values, but each entry in `u` must correspond to a degree of freedom in `dh`,
+see [`write_nodedata`](@ref) for details. This function should be used directly when exporting values already 
+sorted by the nodes in the grid. 
+"""
+function write_solution(vtks::VTKStream, u, suffix="")
+    _vtk_write_solution(vtks.vtk, vtks.grid_or_dh, u, suffix)
+    return vtks
+end
+
+"""
+    write_projected(vtks::VTKStream, proj::L2Projector, vals::Vector, name::AbstractString)
+
+Write `vals` that have been projected with `proj` to the vtk file in `vtks`
+"""
+function write_projected(vtks::VTKStream, proj::L2Projector, vals, name)
+    if getgrid(vtks) !== get_grid(proj.dh)
+        @warn("The grid saved in VTKStream and L2Projector are not aliased, no checks are performed to ensure that they are equal")
+    end
+    data = _evaluate_at_grid_nodes(proj, vals, #=vtk=# Val(true))::Matrix
+    @assert size(data, 2) == getnnodes(get_grid(proj.dh))
+    _vtk_write_nodedata(vtks.vtk, data, name; component_names=get_component_names(eltype(vals)))
+    return vtks
+end
+
+"""
+    write_celldata(vtks::VTKStream, celldata, name)
+
+Write the `celldata` that is ordered by the cells in the grid to the vtk file.
+"""
+function write_celldata(vtks::VTKStream, celldata, name)
+    WriteVTK.vtk_cell_data(vtks.vtk, celldata, name)
+end
+
+"""
+    write_nodedata(vtks::VTKStream, nodedata::Vector{Real}, name)
+    write_nodedata(vtks::VTKStream, nodedata::Vector{<:AbstractTensor}, name)
+    
+Write the `nodedata` that is ordered by the nodes in the grid to the vtk file.
+
+When `nodedata` contains `Tensors.Vec`s, each component is exported. 
+Two-dimensional vectors are padded with zeros.
+
+When `nodedata` contains second order tensors, the index order, 
+`[11, 22, 33, 23, 13, 12, 32, 31, 21]`, follows the default Voigt order in Tensors.jl.
+"""
+function write_nodedata(vtks::VTKStream, nodedata, name)
+    _vtk_write_nodedata(vtks.vtk, nodedata, name)
+end
+
+
+"""
+    write_nodeset(vtks::VTKStream, nodeset::String)
+
+Export nodal values of 1 for nodes in `nodeset`, and 0 otherwise
+"""
+function write_nodeset(vtks::VTKStream, nodeset::String)
+    grid = getgrid(vtks)
     z = zeros(getnnodes(grid))
     z[collect(getnodeset(grid, nodeset))] .= 1.0
-    vtk_point_data(vtk, z, nodeset)
+    WriteVTK.vtk_point_data(vtks.vtk, z, nodeset)
+    return vtks
 end
 
 """
-    vtk_cellset(vtk, grid::Grid)
+    write_cellset(vtks)
```

Comment:
This now becomes somewhat weird looking when you don't pass the grid. Perhaps it is fine to still do that? (And same with dofhandler).

Reply:
Do you mean to allow both not passing and passing the grid? (for `write_celldata`, `write_nodedata`, `write_cellset`, `write_nodeset`, and `write_cell_colors`). 
And allow both not passing and passing the dofhandler with `write_solution`? 
Or alternatively, drop the internal dofhandler, and require that the grid or dofhandler is passed?

Reply:
I was thinking about the latter option. In general I think the grid/dofhandler will be available so you don't really need to pass some extra arguments just to send it to the vtk output function.

Reply:
Currently, `vtk_cell_data`, `vtk_point_data`, and `vtk_cell_data_colors` don't require a `grid` input. The two first relies on `WriteVTK`'s grid representation, and `vtk_cell_data_colors` first use the `Ncls` field of `DatasetFile`. 

By some not-so-nice deprecations, I can still make it backward compatible, but perhaps it is better to just provide an error on `vtk_grid(filename, ::AbstractGrid)` with an update guide instead? (Similar to for CellScalarValues and CellVectorValues)